tools/bpf2c/bpf2c.cpp

```diff
@@ -1,36 +1,120 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+#include <functional>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <regex>
+#include <string>
+#include <tuple>
+#include <vector>
 #include "bpf_code_generator.h"
 
+const char bpf2c_driver[] =
+#include "bpf2c_driver.template"
+    ;
+
+const char bpf2c_dll[] =
+#include "bpf2c_dll.template"
+    ;
+
+void
+emit_driver_skeleton(const std::string& c_name)
+{
+    std::cout << std::regex_replace(bpf2c_driver, std::regex(std::string("___META_DATA_TABLE___")), c_name)
+              << std::endl;
+}
+
+void
+emit_dll_skeleton(const std::string& c_name)
+{
+    std::cout << std::regex_replace(bpf2c_dll, std::regex(std::string("___META_DATA_TABLE___")), c_name) << std::endl;
+}
+
 int
 main(int argc, char** argv)
 {
     try {
+        enum class output_type
+        {
+            Bare,
+            KernelPE,
+            UserPE,
+        } type = output_type::Bare;
+        std::string verifier_output_file;
+        std::string file;
+        std::vector<std::string> sections;
+        std::map<std::string, std::tuple<std::string, std::function<void(std::vector<std::string>::iterator&)>>>
+            options = {
+                {"--sys",
+                 {"Generate code for a Windows driver",
+                  [&](std::vector<std::string>::iterator&) { type = output_type::KernelPE; }}},
+                {"--dll",
+                 {"Generate code for a Windows DLL",
+                  [&](std::vector<std::string>::iterator&) { type = output_type::UserPE; }}},
+                {"--bpf",
+                 {"ELF file containing BPF byte code",
+                  [&](std::vector<std::string>::iterator& it) { file = *(++it); }}},
+                {"--section",
+                 {"List of sections to process ",
```

Comment:
how do you specify a list?  include the option multiple times?   comma separated list?  something else?   Update usage text.